src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs

```diff
@@ -102,46 +102,27 @@ static TimeSpan GetDuration(ToastDuration duration)
 #endif
 
 #if ANDROID || IOS || MACCATALYST || WINDOWS
-	static NativeToast? nativeToast;
-
-	static NativeToast? NativeToast
-	{
-		get
-		{
-			return MainThread.IsMainThread
-				? nativeToast
-				: throw new InvalidOperationException($"{nameof(nativeToast)} can only be called from the Main Thread");
-		}
-		set
-		{
-			if (!MainThread.IsMainThread)
-			{
-				throw new InvalidOperationException($"{nameof(nativeToast)} can only be called from the Main Thread");
-			}
-
-			nativeToast = value;
-		}
-	}
```

Comment:
Can we keep this, replacing `MainThread` with `Dispatcher`? 

We added this after getting random crashes in the sample app that contained cryptic error messages. 

The crashes occurred because `Snackbar` was being modified from a non-UI thread, which makes sense. However, the StackTrace / Error Message provided from OS was very vague and unclear about _why_ the crash happened. 

Long story short, we added this `MainThread.IsMainThread` logic so that when future devs try to consume the `Snackbar` from a background thread will get a clear + concise error message that tells them exactly why it crashed, rather than some confusing OS exception like `SIGSEV blahblahblah`.

Reply:
This code doesn't run on `net6.0`, only on `net6.0-ios`, `net6.0-android`, `net6.0-maccatalyst`, and `net6.0-windows`, so it shouldn't impact our Unit Tests.